### ISSUE TITLE
Wiki Localization

### ISSUE THREAD

**Issue by luii** [NodeOS/NodeOS#222]
Hey,
if its ok i would translate the entire wiki into german.

**Comment by lite20** [comment]
Yes of course! I could help a little bit too. My German isn't 100% but it
could always be revised :D

Now that I think about it, I could do a French translation too as I'm
fluent in French.

I guess just make mirror of the current wiki but in German, and I'll have
the landing page be multilingual. Thanks for the contribution!

**Comment by piranna** [comment]
While you are on it, if you both could review and update the info on the wiki it would be great :-D

**Comment by luii** [comment]
Sure, yesterday i cloned the wiki to my local machine and update some stuff.
One Question: Is it possible to push the wiki to the master branch of the repo?

**Comment by piranna** [comment]
> Is it possible to push the wiki to the master branch of the repo?

What do you mean? GitHub wikis has their own (hidden) git repo, that's the same name of the main project but ending in '.wiki'

https://help.github.com/articles/adding-and-editing-wiki-pages-locally/

Are you talking about this, or about moving the wiki to the `docs` folder of the project?

**Comment by luii** [comment]
Basicly i mean that im rewriting the docs at the moment and when im done i want to push them to the wiki, do i need the permission for this or can anybody push to it.

**Comment by luii** [comment]
I have no permission to push to the wiki :-(

**Comment by piranna** [comment]
> Basicly i mean that im rewriting the docs at the moment and when im done
i want to push them to the wiki, do i need the permission for this or can
anybody push to it.

As far as I know, anyone can publish on the wiki.

**Comment by luii** [comment]
Ok like i said i cloned the wiki to my local machine and i want to push to the wiki but i get a 403 Permission denied

**Comment by luii** [comment]
Maybe you can add me as a member of nodeos

**Comment by piranna** [comment]
> Ok like i said i cloned the wiki to my local machine and i want to push to the wiki but i get a 403 Permission denied

How did you cloned the wiki and how are you doing the push?

**Comment by lite20** [comment]
If I recall correctly, wiki's are just repositories so they can be cloned.
I also could alter it's content without being a member of the repo so maybe
anyone can push too.

**Comment by lite20** [comment]
I already did update most of it :)

I decided I'd be a wiki maintainer since I like to type so I've been
keeping it fairly up to date :P. I just need to write about the logon

**Comment by luii** [comment]
@piranna i cloned by the link that is displayed on the wiki page under the sidebar
and then it gives me a folder called NodeOS.wiki with a .git inside the rest is just git add, comit and push nothing more but like i said above i get a 403

@lite20 currently im updating it make it more user friendly and rearrange some Pages for more readability

**Comment by lite20** [comment]
> currently im updating it make it more user friendly and rearrange some Pages for more readability

Yeah go for it! I don't mind.

**Comment by lite20** [comment]
@luii Email me the next time you're ready to push and I'll see what we can do.

**Comment by luii** [comment]
@lite20 can i become a member of nodeos?

**Comment by piranna** [comment]
> @lite20 can i become a member of nodeos?

Done.

**Comment by luii** [comment]
@lite20 how can i email you, there is no email on your profile

**Comment by lite20** [comment]
Oh you couldn't push? Create a fork?

**Comment by luii** [comment]
already pushed the changes to the wiki

**Comment by lite20** [comment]
> @lite20 <https://github.com/lite20> how can i email you, there is no
email on your profile

Oh that's weird I thought I added it. If anyone ever needs to reach me, you
can use lite260@gmail.com

**Comment by luii** [comment]
Ok that good, if you dont like parts of the new docs please feel free to contact me

**Comment by lite20** [comment]
I'll take a look at them right now. :)

**Comment by luii** [comment]
@lite20 how should i translate the file and where should i place them, in a seperate folder inside the `docs` folder in NodeOS?

**Comment by piranna** [comment]
> @lite20 how should i translate the file and where should i place them, in a seperate folder inside the docs folder in NodeOS?

We have two options, have them in the wiki or move them to the `docs` folder in NodeOS source code. I vote for the second option and deprecate the wiki (I always forget it exists...).

**Comment by luii** [comment]
I would place them in the `docs`, but keep the wiki and i can update it!
Any suggenstions on the folder structure for the `docs`?
I would use ISO 639-1 language codes for the folder names corresponding to the language they'll be translated into.

```
NodeOS
-> docs
---> en - English Translation
---> fr - French Translation
---> de - German Translation
---> es - Spanish Translation
```

**Comment by piranna** [comment]
I was going to propose to follow the schema used by GitBook for the multi-language books, but I think it's the same that you have proposed.

**Comment by Coretool** [comment]
Do we have enough people who speak german and french ? I speak both... so I can help if needed ...

**Comment by piranna** [comment]
I can be able only english and spanish.

**Comment by luii** [comment]
I translate the current Wiki to german

**Comment by Coretool** [comment]
I can do frech ...

**Comment by luii** [comment]
@piranna Should we push directly to the Repo with [ci skip] in the commit or should we create a seperate repo (like: "nodeos-de") and link it with git submodule to this repo?

**Comment by piranna** [comment]
I'm against to move the docs to independent repos, I think it's better to have them here with the main source code. What's that thing of `[ci skip]`?

**Comment by lite20** [comment]
> Do we have enough people who speak german and french ? I speak both...

What really? I'm multilingual too! French is actually my dominant language.

**Comment by lite20** [comment]
@piranna we could create a mirror site like we did for the blog to hide the bloat of all the other languages being present? We could also then format it as we pleased.

**Comment by luii** [comment]
@piranna if you place `[ci skip]` or `[skip ci]` inside the commit title semaphoreci will skip this commit
because its not a build process relevant commit.

@lite20 would it not be better to push it all to the docs folder and then create a list of languages inside the wiki?

**Comment by Coretool** [comment]
@lite20 Oh wow ^^ Yeah... I'm swiss ... we've got about 4 official languges and even more that we (can) learn in school ^^ 

@luii I like the idea of seperated sites bette, I think it'd be much cleaner :/

**Comment by luii** [comment]
@Coretool i also prefer the idea of seperated stites.
I would also create git submodules in the nodeos/docs folder

**Comment by piranna** [comment]
> @piranna we could create a mirror site like we did for the blog to hide
the bloat of all the other languages being present? We could also then
format it as we pleased.

I think that's diferent, the blog was an independent project. What's the
standard on Node.js projects beside Readme.md? A 'doc' folder?

**Comment by piranna** [comment]
> @piranna if you place [ci skip] or [skip ci] inside the commit title
semaphoreci will skip this commit
> because its not a build process relevant commit.

Good trick, didn't know it :-)

> @lite20 would it not be better to push it all to the docs folder and then
create a list of languages inside the wiki?

I agree on that, only that put the languages list on the repo too.

**Comment by piranna** [comment]
> I would also create git submodules in the nodeos/docs folder

I don't like submodules too much... :-/

**Comment by lite20** [comment]
Check out how I handled double languages with HexCade:
https://bitbucket.org/Lite20/hexcade/wiki/Home

We could potentially do this?

**Comment by piranna** [comment]
[GitBook multi-language format](http://toolchain.gitbook.com/languages.html) is very similar to the solution that you propose :-) This way we could be able later easily publish NodeOS docs too :-D

**Comment by lite20** [comment]
GitBook sounds good to me! Let's do it! I'll probably implement it w/ HexCade too.

By the way, since I know you're curious, HexCade is an interpreted language I started creating to better understand interpreted languages. It's key features are how it handles threads, it's true asynchronous nature, and it's object context design (in JS we have contexts; in HxC, contexts from, for example functions, are variables too and can be accessed by other parts of a program if needed)
